test.js

```diff
@@ -205,3 +205,25 @@ test('Should strip the url to only match the pathname', t => {
 
   instance.run(req, res)
 })
+
+test('should keep the context', t => {
+  t.plan(6)
+
+  const instance = middie(function (err, a, b) {
```

Comment:
I think we should change this function signature to `(err, req, res, ctx)`, so that we can completely avoid the `.call()`  and assigning `this`. It would be faster.